satellite/accounting/rollup/rollup.go

```diff
@@ -64,44 +67,54 @@ func (r *Service) Close() error {
 // Rollup aggregates storage and bandwidth amounts for the time interval.
 func (r *Service) Rollup(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
-	// only Rollup new things - get LastRollup
-	lastRollup, err := r.sdb.LastTimestamp(ctx, accounting.LastRollup)
-	if err != nil {
-		return Error.Wrap(err)
-	}
-	// unexpired orders with created at times before the last rollup timestamp could still have been added later
-	if !lastRollup.IsZero() {
-		lastRollup = lastRollup.Add(-r.OrderExpiration)
-	}
 
-	rollupStats := make(accounting.RollupStats)
-	latestTally, err := r.RollupStorage(ctx, lastRollup, rollupStats)
+	nodeIDs, err := r.overlay.GetAllNodeIDs(ctx)
 	if err != nil {
 		return Error.Wrap(err)
 	}
 
-	err = r.RollupBW(ctx, lastRollup, rollupStats)
-	if err != nil {
-		return Error.Wrap(err)
-	}
+	var overallLatestTally time.Time
+	for _, id := range nodeIDs {
+		nodeLastRollup, err := r.sdb.NodeLastTimestamp(ctx, id, accounting.LastRollup)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+		// unexpired orders with created at times before the last rollup timestamp could still have been added later
+		if !nodeLastRollup.IsZero() {
+			nodeLastRollup = nodeLastRollup.Add(-r.OrderExpiration)
+		}
 
-	// remove the latest day (which we cannot know is complete), then push to DB
-	latestTally = time.Date(latestTally.Year(), latestTally.Month(), latestTally.Day(), 0, 0, 0, 0, latestTally.Location())
-	delete(rollupStats, latestTally)
-	if len(rollupStats) == 0 {
-		r.logger.Info("RollupStats is empty")
-		return nil
-	}
+		rollupStats := make(accounting.RollupStats)
+		latestTally, err := r.RollupStorage(ctx, id, nodeLastRollup, rollupStats)
+		if err != nil {
+			return Error.Wrap(err)
+		}
 
-	err = r.sdb.SaveRollup(ctx, latestTally, rollupStats)
-	if err != nil {
-		return Error.Wrap(err)
+		err = r.RollupBW(ctx, id, nodeLastRollup, rollupStats)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+
+		// remove the latest day (which we cannot know is complete), then push to DB
+		latestTally = time.Date(latestTally.Year(), latestTally.Month(), latestTally.Day(), 0, 0, 0, 0, latestTally.Location())
+		delete(rollupStats, latestTally)
+		if len(rollupStats) == 0 {
+			r.logger.Info("RollupStats is empty")
+			return nil
+		}
+
+		err = r.sdb.SaveRollup(ctx, latestTally, rollupStats)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+
+		overallLatestTally = latestTally
```

Comment:
shouldn't this line only happen if overallLatestTally is less than latestTally?

Reply:
actually, see comment below, i think we should have a perNodeOldestLatestTally or something, right?

Reply:
I think if we just get the overall oldest latest tally (LOL) it would work, and we only need to delete once. But as Ethan said in the description, maybe we don't want to delete tallies anyway

Reply:
okay so overallOldest instead of overallLatest, sure